docs/docs/creating-a-source-plugin.md

```diff
@@ -18,147 +18,469 @@ At a high-level, a source plugin:
+  const { createTypes } = actions
+  createTypes(`
+    type Post implements Node {
+      id: ID!
+      # create a relationship between Post and the File nodes for optimized images
+      author: Author @link(from: "author.name" by: "name") // highlight-line
+      # ... other fields
+    }`)
+}
+```
 
 ##### Creating the reverse relationship
 
-It's often convenient for querying to add to the schema backwards references. For example, you might want to query the Author of a Post but you might also want to query all the posts an author has written.
+It's often convenient for querying to add to the schema backwards references. For example, you might want to query the author of a post, but you might also want to query all the posts an author has written.
 
-If you want to call this field on `Author` `posts`, you would create a field called `posts___NODE` to hold the relationship to Posts. The value of this field should be an array of Post IDs.
+If you want to call a field to access the author on the `Post` nodes using the inference method, you would create a field called `posts___NODE` to hold the relationship to posts. The value of this field should be an array of `Post` IDs.
 
 Here's an example from the [WordPress source plugin](https://github.com/gatsbyjs/gatsby/blob/1fb19f9ad16618acdac7eda33d295d8ceba7f393/packages/gatsby-source-wordpress/src/normalize.js#L178-L189).
 
+With schema customization, you would add the `@link` directive to your Author type. The `@link` directive will look for an ID on the `post` field of the Author nodes, which can be added when the Author nodes are created.
+
+```javascript:title=source-plugin/gatsby-node.js
+exports.createSchemaCustomization = ({ actions }) => {
+  const { createTypes } = actions
+  createTypes(`
+    type Post implements Node {
+      id: ID!
+      # create a relationship between Post and the File nodes for optimized images
+      author: Author @link(from: "author.name" by: "name") // highlight-line
+      # ... other fields
+    }
+    
+    type Author implements Node {
+      name: String!
+      post: Post @link // highlight-line
+    }`)
+}
+```
+
+You can read more about connecting foreign key fields with schema customization in the guide on [customizing the GraphQL schema](/docs/schema-customization/#foreign-key-fields).
+
+#### Option 2: transformation relationships
+
+When a node is _completely_ derived from another node you'll want to use a transformation relationship. An example that is common in source plugins is for transforming File nodes from remote sources, e.g. images. You can read about this use case in the section below on [sourcing images from remote locations](/docs/creating-a-source-plugin/#sourcing-images-from-remote-locations).
+
+You can find more information about transformation relationships in the [creating a transformer plugin guide](/docs/creating-a-transformer-plugin/#creating-the-transformer-relationship).
+
 #### Union types
```

Comment:
Are union types another option? Additional? A bit confusing in the header hierarchy.

Reply:
They are additional, but I get your point that it's confusing yeah. I updated the text under it to explain it a little more, but I think it's a less common thing so didn't want to put it up in the section on adding relationships when it felt a little more advanced. 

It does correspond to both types of relationships though.